### PR TITLE
Put/Patch request

### PR DESCRIPTION
**Pull request by matthewleemle**
Only problem is no safeguard whether the requested added user is already in the added responders (likely not a major issue)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,6 +183,38 @@ HTTP Response:
...
+## Putting Help Request
+
+PUT "/help-requests"
+
+Request Body:
+
+```
+{
+	"helpReqId": "somehelpreqid",
+	"newResponder": "someresponderid",
```
Comment (Asiapenguin) [inline README.md]: newResponderId

File: models/help_request.js [hunk 2]
```diff
@@ -2,23 +2,33 @@ var database = require("../database/database");
 var mongoose = database.getmongoose();
 var Schema = mongoose.Schema;
 
+const arrayLimit = (val) => {
+  return val.length <= 6;
```
Comment (Asiapenguin) [inline models/help_request.js]: Somehow it's letting me add 7 entries

File: controllers/help_request.js [hunk 3]
```diff
@@ -5,6 +5,45 @@ var UserModel = require("../models/user").model;
...
+        _id: new ObjectId(helpReqId)
+      })
+    } catch (err) {
+      handle.badRequest(res, err.message);
+    }
+
+    if (help_request==null)
+      handle.badRequest(res,"Help Request does not exist")
+    else{
+      help_request.responderIds.push({_id:false, id: newResponder});
```
Comment (Asiapenguin) [inline controllers/help_request.js]: Please check if the id exists in the array first before pushing to it

File: controllers/help_request.js [hunk 4]
```diff
@@ -5,6 +5,45 @@ var UserModel = require("../models/user").model;
 var NotificationService = require("../services/notification.service");
 var UserService = require("../services/user.service");
 
+const putHelpRequest = async (req, res) => {
+  let status = req.body.status;
+  let newResponder = req.body.newResponder;
```
Comment (Asiapenguin) [inline controllers/help_request.js]: if this is not supplied, the function will push {} to the array of responderIds of the help request

File: models/help_request.js (hunk 2 above)
Comment (Asiapenguin) [reply models/help_request.js]: And this doesnt actually send an error as a response. It still sends back 200 but logs the error. Maybe we need to find a new way of handling this

File: README.md [hunk 5]
```diff
@@ -183,6 +183,38 @@ HTTP Response:
...
+}
+```
+
+HTTP Response:
+
+```
+{
+    "id": string,
+    "userId": string,
+    "responderId": string || null (initialized, only filled when a responder accepts this request),
```
[the remark this answers is not in the document]
Comment (matthewleemle) [reply README.md]: Yeah, I just forgot the update the README

File: models/help_request.js (hunk 2 above)
Comment (matthewleemle) [reply models/help_request.js]: Does it add 7 entries in the database or does it just not throw an error in the http response?

Comment (matthewleemle) [reply models/help_request.js]: OK it seems to be just not throwing an error but doesnt actually add it

**Comment by matthewleemle** [comment]
@Asiapenguin please check the changes I made

File: controllers/help_request.js (hunk 4 above)
Comment (matthewleemle) [reply controllers/help_request.js]: fixed

File: controllers/help_request.js (hunk 3 above)
Comment (matthewleemle) [reply controllers/help_request.js]: Changed appropriately

File: models/help_request.js (hunk 2 above)
Comment (matthewleemle) [reply models/help_request.js]: Changed appropriately

File: README.md [hunk 6]
```diff
@@ -183,6 +183,38 @@ HTTP Response:
     "updatedAt": date
 }
 ```
+
+## Putting Help Request
+
+PUT "/help-requests"
```
[the remark this answers is not in the document]
Comment (matthewleemle) [reply README.md]: Changed appropriately

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (matthewleemle) [reply README.md]: Updated appropriately

**Pull request merged**